testdata/dnn/download_models.py

```diff
@@ -780,13 +786,14 @@ def save(self, r):
 # Note: models will be downloaded to current working directory
 #       expected working directory is opencv_extra/testdata/dnn
 if __name__ == '__main__':
-    failedSums = []
+    failedModels = []
     for m in models:
         print(m)
         if not m.get():
-            failedSums.append(m.filename)
-    if failedSums:
-        print("Checksum verification failed for:")
-        for f in failedSums:
+            failedModels.append(m.filename)
```

Comment:
```
for m in models:
    print(m)
    try:
        if not m.get():
            failedModels.append(m.filename)
    except Exception as e:
        print('Exception for model {}:\n{}'.format(m.filename, e))
        failedModels.append(m.filename)
```

Reply:
The idea is to catch only exceptions which are caused by specific runtime problems: file can not be opened, server does not respond, etc.. All other exceptions which can be caused by broken script or system misconfiguration should be raised and break the program execution.

Reply:
> other exceptions which can be caused by broken script

so scripts would fail faster (and still print "unexpected" problems).

We may filter by exception type, but it is **not reliable**:
- `OSError`
- `urllib.error.URLError`
- `urllib.error.HTTPError`
- something else...

Reply:
> so scripts would fail faster (and still print "unexpected" problems).
It will not fail faster, for example if some of models in the config have missing attribute, this model will fail, but download process will continue. User will need to analyze logs each time to determine what caused the problem: wrong configuration or runtime issue.

Reply:
Perhaps, we might try `AssertionError` separate handling (with loop break) instead.